app/Commands/Sites/CreateCommand.php

```diff
@@ -70,58 +46,76 @@ protected function action(): int
             return self::INVALID;
         }
 
-        $server    = $this->selectServer('deploy to')->first();
-        $userInput = $this->getUserInput();
-        $site      = $this->spinupwp->createSite($server->id, array_merge($this->arguments(), $this->options(), $userInput));
+        $server       = $this->selectServer('deploy to')->first();
+        $this->domain = $this->option('domain') ?? $this->resolveAnswer(['prompt' => 'Domain Name'], $this->nonInteractive());
+        $userInput    = $this->promptForAnswers($this->nonInteractive());
+
+        $site = $this->spinupwp->createSite($server->id, array_merge($this->arguments(), $this->options(), ['domain' => $this->domain], $userInput));
 
         $this->successfulStep("{$site->domain} is {$site->status} (event_id = {$site->eventId()})");
 
         return self::SUCCESS;
     }
 
-    /**
-     * Use different options depending on site install type.
-     */
-    protected function getAvailableOptionIO(string $type): array
+    protected function getPrompts(): array
     {
-        switch ($type) {
+        $prompts = [
```

Comment:
What, specifically, are we defining here? What is a `prompt`? Is this part but not all of a bigger picture concept we could expand this config to include?

Reply:
✅ re: our previous convo, sticking with "prompt" here. We're defining the question "prompts" and returning the answers.